packages/core/src/routes/session.ts

```diff
@@ -28,28 +28,118 @@ import { AnonymousRouter } from './types';
 
 export default function sessionRoutes<T extends AnonymousRouter>(router: T, provider: Provider) {
   router.post(
-    '/session',
+    '/session/sign-in/username-and-password',
+    koaGuard({ body: object({ username: string(), password: string() }) }),
+    async (ctx, next) => {
+      const {
+        prompt: { name },
+      } = await provider.interactionDetails(ctx.req, ctx.res);
+
+      if (name === 'consent') {
+        ctx.body = { redirectTo: ctx.request.origin + '/session/consent' };
+
+        return next();
+      }
+
+      if (name === 'login') {
+        const { username, password } = ctx.guard.body;
+
+        if (username && password) {
+          await signInWithUsernameAndPassword(ctx, provider, username, password);
+        } else {
+          throw new RequestError('session.insufficient_info');
+        }
+
+        return next();
+      }
+
+      throw new errors.InvalidRequest(`Prompt not supported: ${name}`);
+    }
+  );
+
+  router.post(
+    '/session/sign-in/phone-and-code',
+    koaGuard({ body: object({ phone: string(), code: string().optional() }) }),
+    async (ctx, next) => {
+      const {
+        // Interaction's JWT identity: jti
+        // https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#user-flows
+        // https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.7
+        jti,
+        prompt: { name },
+      } = await provider.interactionDetails(ctx.req, ctx.res);
+
+      if (name === 'consent') {
+        ctx.body = { redirectTo: ctx.request.origin + '/session/consent' };
+
+        return next();
+      }
+
+      if (name === 'login') {
+        const { phone, code } = ctx.guard.body;
+
+        if (phone && !code) {
+          await sendSignInWithPhonePasscode(ctx, jti, phone);
+        } else if (phone && code) {
+          await signInWithPhoneAndPasscode(ctx, provider, { jti, phone, code });
+        } else {
+          throw new RequestError('session.insufficient_info');
+        }
+
+        return next();
+      }
+
+      throw new errors.InvalidRequest(`Prompt not supported: ${name}`);
+    }
+  );
+
+  router.post(
+    '/session/sign-in/email-and-code',
```

Comment:
per discussion the route should be divided by functionality instead of flow type?